packages/cli/src/commands/account/list.ts

```diff
@@ -8,24 +8,32 @@ export default class AccountList extends BaseCommand {
 
   static flags = {
     ...BaseCommand.flags,
+    local: flags.boolean({
+      default: false,
+      description: 'If set, only show local and hardware wallet accounts',
+    }),
   }
 
   requireSynced = false
 
   async run() {
-    this.parse(AccountList)
+    const res = this.parse(AccountList)
 
     const celoProvider: CeloProvider = this.kit.web3.currentProvider as any
-    const addresses = await this.kit.web3.eth.getAccounts()
+
+    // Retreive accounts from the connected Celo node.
+    const addresses = []
+    if (res.flags.local) {
+      addresses.push(...(await this.kit.web3.eth.getAccounts()))
+    }
+
+    // Get addresses from the local wallet.
     const localAddresses = celoProvider.wallet
       .getAccounts()
       .map((value) => toChecksumAddress(value))
-    let localName = 'Local'
-    const res: ParserOutput<any, any> = this.parse()
 
-    if (res.flags.useLedger) {
-      localName = 'Ledger'
-    }
+    // Display the addresses.
+    const localName = res.flags.useLedger ? 'Ledger' : 'Local'
     console.log('All Addresses: ', addresses)
```

Comment:
This line shows every address, and with this you are showing [] for local

What you could do, is to change the order, retrieve first the local wallet addresses, and then all of them.
And instead of the last recommendation add this:

```
    // Get addresses from the local wallet.
    const localAddresses = celoProvider.wallet
      .getAccounts()
      .map((value) => toChecksumAddress(value))

   const addresses = res.flags.local? localAddresses : await this.kit.web3.eth.getAccounts()
```

Reply:
Does https://github.com/celo-org/celo-monorepo/pull/3830/commits/c24e42aa5cf206d410b7c301b8513935ed585699 address this comment?